service/matching/version_rule_test.go

```diff
@@ -389,16 +436,104 @@ func TestDeleteAssignmentRuleTestRequireUnfiltered(t *testing.T) {
 	assert.Equal(t, 0, len(getListResp(t, data).GetAssignmentRules()))
 }
 
-// Test requirement that number of rules does not exceed max rules, including that deleting a rule fixes the error.
+// Test requirement that number of rules does not exceed max rules.
 func TestInsertRedirectRuleMaxRules(t *testing.T) {
+	t.Parallel()
```

Comment:
This test looks similar to InsertAssignmentRuleMaxRules. The difference is the rule and the insert function. Is that right?

Reply:
yes exactly. I don't think it's worth making a generic framework to build both of those tests, because we wouldnt reuse it for the other requirements. `MaxRules` is the only requirement that is the same for both Redirect Rules and Assignment Rules, so this is the only test that is going to be this similar across the two rule types.